content_sync/apis/github.py

```diff
@@ -139,28 +156,41 @@ def create_branch(self, branch: str, source: str, delete_source=False) -> Branch
 
     @retry_on_failure
     def upsert_content_file(
-        self, website_content: WebsiteContent, message: str, **kwargs
-    ) -> Commit:
+        self, website_content: WebsiteContent, **kwargs
+    ) -> Optional[Commit]:
         """
         Create or update a file in git.
         """
-        if not website_content.content_filepath:
-            # No filepath, nothing to do"
+        destination_filepath = get_destination_filepath(
+            website_content, self.site_config
+        )
+        if not destination_filepath:
+            # No filepath, nothing to do
```

Comment:
I added error logging for the situations where `get_destination_filepath` could still return None. Those situations shouldn't happen in practice, but it's still theoretically possible, and if nothing else it will help to catch those situations for local development while we hack on different site config structures